var/spack/repos/builtin/packages/sqlite/package.py

```diff
@@ -75,3 +88,11 @@ def configure_args(self):
             args.append('--build=powerpc64le-redhat-linux-gnu')
 
         return args
+
+    @run_after('install')
+    def build_libsqlitefunctions(self):
+        if '+functions' in self.spec:
+            gcc = which('gcc')
+            gcc('-fPIC', '-lm', '-shared', 'extension-functions.c', '-o',
```

Comment:
Use `self.compiler.pic_flag` to make this generic enough to work for other compilers.